app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/view/CharacterView.kt

```diff
@@ -1,19 +1,22 @@
-package com.puzzlebench.clean_marvel_kotlin.presentation.mvp
+package com.puzzlebench.clean_marvel_kotlin.presentation.mvp.view
 
 import android.support.v7.widget.GridLayoutManager
 import android.view.View
 import com.puzzlebench.clean_marvel_kotlin.R
-import com.puzzlebench.cmk.domain.model.Character
 import com.puzzlebench.clean_marvel_kotlin.presentation.MainActivity
 import com.puzzlebench.clean_marvel_kotlin.presentation.adapter.CharacterAdapter
 import com.puzzlebench.clean_marvel_kotlin.presentation.extension.showToast
-import kotlinx.android.synthetic.main.activity_main.*
+import com.puzzlebench.clean_marvel_kotlin.presentation.mvp.CharacterFragment
+import com.puzzlebench.clean_marvel_kotlin.presentation.util.ONE
+import com.puzzlebench.cmk.domain.model.Character
+import kotlinx.android.synthetic.main.activity_main.progressBar
+import kotlinx.android.synthetic.main.activity_main.recycleView
 import java.lang.ref.WeakReference
 
 class CharacterView(activity: MainActivity) {
     private val activityRef = WeakReference(activity)
-    private val SPAN_COUNT = 1
-    var adapter = CharacterAdapter { character -> activity.applicationContext.showToast(character.name) }
+    private val SPAN_COUNT = ONE.toInt()
+    var adapter = CharacterAdapter { character -> showFragmentDialog(character) }
 
     fun init() {
         val activity = activityRef.get()
```

Comment:
please use kotlin style code.. 
activityRef.get()?.let {
     activity.recycleView.layoutManager = GridLayoutManager(activity, SPAN_COUNT)
     activity.recycleView.adapter = adapter
     showLoading()
}

Reply:
okey, got it!